source/solvers/cahn_hilliard.cc

```diff
@@ -1163,14 +1206,16 @@ CahnHilliard<dim>::calculate_barycenter(const GlobalVectorType &solution,
 
           for (unsigned int q = 0; q < n_q_points; q++)
             {
-              const double JxW = fe_values_cahn_hilliard.JxW(q);
+              const double JxW          = fe_values_cahn_hilliard.JxW(q);
+              const double phase_values = phase_cahn_hilliard_values[q];
+
 
 
-              volume += (1 - phase_cahn_hilliard_values[q]) * 0.5 * JxW;
-              barycenter_location += (1 - phase_cahn_hilliard_values[q]) * 0.5 *
-                                     quadrature_locations[q] * JxW;
-              barycenter_velocity += (1 - phase_cahn_hilliard_values[q]) * 0.5 *
-                                     velocity_values[q] * JxW;
+              volume += (1 - phase_values) * 0.25 * JxW;
+              barycenter_location +=
+                (1 - phase_values) * 0.25 * quadrature_locations[q] * JxW;
+              barycenter_velocity +=
+                (1 - phase_values) * 0.25 * velocity_values[q] * JxW;
```

Comment:
You're right, it should be 0.5 and not 0.25, I did a change of variable, but it is not correct here. There is a factor because in the fluid 1, which have phase_value = -1, the integrand equals to 2. In the fluid 0 it is equal to 0

Reply:
Strangely, it did not have any noticeable effect on the examples which were using this functionality after double checking